internal/portforward/service.go

```diff
@@ -175,6 +176,15 @@ func (s *Service) validateCreateRequest(r CreateRequest) error {
 	return nil
 }
 
+func (s *Service) FindPodSpecForService(ctx context.Context, apiVersion, kind, namespace, name string) (*corev1.PodSpec, error) {
+	pod, err := s.findPodForService(ctx, apiVersion, kind, namespace, name)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pod.Spec, nil
+}
+
```

Comment:
It doesn't look like this is ever used inside or outside of the package, remove.